inception/inception-project/src/main/java/de/tudarmstadt/ukp/clarin/webanno/project/ProjectServiceImpl.java

```diff
@@ -955,4 +948,75 @@ public void setProjectState(Project aProject, ProjectState aState)
                     .publishEvent(new ProjectStateChangedEvent(this, aProject, oldState));
         }
     }
+
+    @Override
+    public String deriveSlugFromName(String aName)
```

Comment:
Optional<String>?

Reply:
Well, not sure. The function returns null if the input is null. I was considering to throw an exception if the input is null, but for chaining it seemed better to pass the null through - so it doesn't really feel like an optional.